applications/RANSApplication/python_scripts/formulations/utilities.py

```diff
@@ -0,0 +1,244 @@
+    if (not RansVariableUtilities.IsAnalysisStepCompleted(
+            model_part, "CONDITION_NORMAL_CALCULATION")):
+
+        # this calculates normals on whole model part, and assigns
+        # NORMAL variable in NodalSolutionStepDataValue container.
+        # NORMAL on conditions is required by some formulations such as inlet condition for
+        # incompressible potential flow velocity formulation, and all boundaries for incompressible
+        # potential flow pressure formulation.
+        Kratos.NormalCalculationUtils().CalculateOnSimplex(
+            model_part.Conditions, domain_size)
+        RansVariableUtilities.AddAnalysisStep(model_part,
+                                              "CONDITION_NORMAL_CALCULATION")
+
+        # This reverts incorrectly calculated nodal NORMALS from previous method
+        # since it spreads condition NORMAL to all nodes of model part, but from this
+        # method, it again spreads condition NORMALs to nodes where condition is applied
+        # with SLIP flag.
+        RansVariableUtilities.AssignConditionVariableValuesToNodes(
+            model_part, Kratos.NORMAL, Kratos.SLIP)
+
+        Kratos.Logger.PrintInfo("NormalCalculationUtils",
+                                "Condition normals calculated.")
+
+
+def CreateRansFormulationModelPart(formulation, element_name, condition_name):
+    formulation.domain_size = formulation.GetBaseModelPart().ProcessInfo[
+        Kratos.DOMAIN_SIZE]
+
+    element_suffix = str(
+        formulation.domain_size) + "D" + str(formulation.domain_size + 1) + "N"
+    condition_suffix = str(formulation.domain_size) + "D" + str(
+        formulation.domain_size) + "N"
+
+    element_name = element_name + element_suffix
+    condition_name = condition_name + condition_suffix
+    return CreateDuplicateModelPart(
+        formulation.GetBaseModelPart(),
+        formulation.GetName() + "_" + element_name + "_" + condition_name,
+        element_name, condition_name, "")
+
+
+def IsBufferInitialized(formulation):
+    return (formulation.GetBaseModelPart().ProcessInfo[Kratos.STEP] + 1 >=
+            formulation.GetMinimumBufferSize())
+
+
+def InitializePeriodicConditions(base_model_part, model_part, variables_list):
+    properties = model_part.CreateNewProperties(
+        model_part.NumberOfProperties() + 1)
+    pcu = KratosCFD.PeriodicConditionUtilities(
+        model_part, model_part.ProcessInfo[Kratos.DOMAIN_SIZE])
+    for variable in variables_list:
+        pcu.AddPeriodicVariable(properties, variable)
+
+    index = model_part.NumberOfConditions()
+    for condition in base_model_part.Conditions:
+        if condition.Is(Kratos.PERIODIC):
+            index += 1
+            node_id_list = [node.Id for node in condition.GetNodes()]
+            periodic_condition = model_part.CreateNewCondition(
+                "PeriodicCondition", index, node_id_list, properties)
+            periodic_condition.Set(Kratos.PERIODIC)
```

Comment:
I think this is independent of the RANS stuff. Shouldn't we add it to the `Initialize` of the base fluid solver?